analyzers/src/SonarAnalyzer.CSharp/Rules/ObjectCreatedDropped.cs

```diff
@@ -25,31 +25,25 @@
 using Microsoft.CodeAnalysis.Diagnostics;
 using SonarAnalyzer.Helpers;
 
-namespace SonarAnalyzer.Rules.CSharp
+namespace SonarAnalyzer.Rules.CSharp;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class ObjectCreatedDropped : SonarDiagnosticAnalyzer
 {
-    [DiagnosticAnalyzer(LanguageNames.CSharp)]
-    public sealed class ObjectCreatedDropped : SonarDiagnosticAnalyzer
-    {
-        internal const string DiagnosticId = "S1848";
-        private const string MessageFormat = "Either remove this useless object instantiation of class '{0}' or use it.";
+    private const string DiagnosticId = "S1848";
+    private const string MessageFormat = "Either remove this useless object instantiation of class '{0}' or use it.";
 
-        private static readonly DiagnosticDescriptor rule =
-            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
+    private static readonly DiagnosticDescriptor Rule = DescriptorFactory.Create(DiagnosticId, MessageFormat);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
 
-        protected override void Initialize(SonarAnalysisContext context)
-        {
-            context.RegisterSyntaxNodeActionInNonGenerated(
-                c =>
+    protected override void Initialize(SonarAnalysisContext context) =>
+        context.RegisterSyntaxNodeActionInNonGenerated(c =>
+            {
+                if (c.Node is ObjectCreationExpressionSyntax creation && creation.Parent is ExpressionStatementSyntax)
```

Comment:
To bump the coverage. We don't need to test for the type, because we register only for that one.
```suggestion
                var creation = (ObjectCreationExpressionSyntax)c.Node;
                if (creation.Parent is ExpressionStatementSyntax)
```